exporter/azuremonitorexporter/logexporter_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azuremonitorexporter
+
+/*
+Contains tests for logexporter.go and log_to_envelope.go
+*/
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/microsoft/ApplicationInsights-Go/appinsights/contracts"
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/model/otlp"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.uber.org/zap"
+)
+
+const (
+	defaultEnvelopeName = "Microsoft.ApplicationInsights.Message"
+	defaultdBaseType    = "MessageData"
+)
+
+var (
+	testLogs = []byte(`{"resourceLogs":[{"resource":{"attributes":[{"key":"service.name","value":{"stringValue":"dotnet"}}]},"instrumentationLibraryLogs":[{"instrumentationLibrary":{},"logs":[{"timeUnixNano":"1643240673066096200","severityText":"Information","name":"FilterModule.Program","body":{"stringValue":"Message Body"},"flags":1,"traceId":"7b20d1349ef9b6d6f9d4d1d4a3ac2e82","spanId":"0c2ad924e1771630"}]}]}]}`)
+)
+
+// Tests proper wrapping of a log record to an envelope
+func TestLogRecordToEnvelope(t *testing.T) {
+	logRecord := getTestLogRecord(t)
+	logPacker := getLogPacker()
+	envelope := logPacker.LogRecordToEnvelope(logRecord)
+
+	assert.NotNil(t, envelope)
```

Comment:
Consider using `require` instead of `assert` for checks for `NotNil` since assert won't stop the test being run.

Require will do a call to `tb.FailNow` which should help prevent any flakiness in future.